kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/string/ends.kt

```diff
@@ -0,0 +1,14 @@
+package io.kotest.matchers.string
+
+import io.kotest.matchers.should
+import io.kotest.matchers.shouldNot
+
+infix fun <A : CharSequence> A?.shouldEndWith(suffix: String): A? {
+   this.toString() should endWith(suffix)
```

Comment:
`null.toString()`

i start to wonder if we should change `endWith` and other `String` matchers to work based on `A : CharSequence` now? i.e.

```kotlin

fun <A : CharSequence> beEmpty() = neverNullMatcher<A> { value ->
   MatcherResult(
      value.toString().isEmpty(),
      "${value.toString().show().value} should be empty",
      "${value.toString().show().value} should not be empty"
   )
}

```